backend/game/cards/keep_cards/turn_manipulation_cards/GiantBrain.py

```diff
@@ -0,0 +1,15 @@
+from game.cards.keep_card import KeepCard
+
+
+class GiantBrain(KeepCard):
+    def __init__(self):
+        super().__init__("Giant Brain", 5, "You have 1 extra die roll each turn")
+
+    def immediate_effect(self, player_that_bought_the_card, other_players):
```

Comment:
This is bad as it has a side-effect of overloading the Keep card method for immediate_effect/discard_effect (aka adding to the deck and removing from the deck).  We will need to verify this is not done on other cards.  For now, remove it from this child card.  The special effect overload method is fine and appropriate though